views/install.ps1.erb

```diff
@@ -0,0 +1,191 @@
+            $hash.Hash = ([BitConverter]::ToString($c.ComputeHash($in))).Replace("-", "").ToUpper()
+          }
+        }
+      }
+      new-object PSObject -Property $hash
+    }
+  } 
+  
+  # PowerShell is only on Windows ATM
+  $platform = 'windows'
+  Write-Verbose "Platform: $platform"
+  
+  # TODO: No Win10 build endpoint yet
+  switch -regex ((get-wmiobject win32_operatingsystem).version) {
+    '10\.0\.\d+' {$platform_version = '2012r2'}
+    '6\.3\.\d+'  {$platform_version = '2012r2'}
+    '6\.2\.\d+'  {$platform_version = '2012'}
+    '6\.1\.\d+'  {$platform_version = '2008r2'}
+    '6\.0\.\d+'  {$platform_version = '2008'}
+  }
+  Write-Verbose "Platform Version: $platform_version"
+  
+  # TODO: When we ship a 64 bit ruby for Windows.
+  $machine = 'x86_64'
+  Write-Verbose "Machine: $machine"
+  
+  Write-Verbose "Project: $project"
+  switch ($project) {
+    'chef'      {$metadata_base_url = "metadata"}
+    'angrychef' {$metadata_base_url = "metadata-angrychef"}
+    'server'    {$metadata_base_url = "metadata-server"} 
+    'chefdk'    {$metadata_base_url = "metadata-chefdk"}
+    'container' {$metadata_base_url = "metadata-container"}
+  }
+  $metadata_array = ("?v=$($version)", 
+    "prerelease=$($prerelease.tostring().tolower())",
+    "nightlies=$($nightlies.tostring().tolower())",
+    "p=$platform",
+    "pv=$platform_version",
+    "m=$machine")
+  $metadata_base_url += [string]::join('&', $metadata_array)
+  $metadata_url = new-uri $base_server_uri $metadata_base_url
+  
+  
+  Write-Verbose "Downloading $project details from $metadata_url"
+  $package_metadata = (Get-WebContent $metadata_url).trim() -split '\n' | 
+    foreach { $hash = @{} } {$key, $value = $_ -split '\s+'; $hash.Add($key, $value)} {$hash}
+  
+  Write-Verbose "Project details: "
+  foreach ($key in $package_metadata.keys) {
+    Write-Verbose "`t$key = $($package_metadata[$key])"
+  }
+  
+  if (-not [string]::IsNullOrEmpty($filename)) {
+    $download_directory = split-path $filename
+    $filename = split-path $filename -leaf
+    if ([string]::IsNullOrEmpty($download_directory)) {
+      $download_directory = $pwd
+    }
+  }
+  else {
+    $filename = ($package_metadata.url -split '/')[-1]
+  }
+  Write-Verbose "Download directory: $download_directory"
+  Write-Verbose "Filename: $filename"
+  
+  if (-not (test-path $download_directory)) {
+    mkdir $download_directory
+  }
+  $download_directory = (resolve-path $download_directory).providerpath
+  $download_destination = join-path $download_directory $filename
+  
+  Write-Verbose "Downloading $project from $($package_metadata.url) to $download_destination."
+  Get-WebContent $package_metadata.url -filepath $download_destination
+  
+  $downloaded_md5 = (get-filehash -path $download_destination -algorithm MD5).Hash.ToLower()
+  $downloaded_sha256 = (get-filehash -path $download_destination -algorithm SHA256).Hash.ToLower()
+  $Valid = ($package_metadata.md5 -eq $downloaded_md5) -and
```

Comment:
If I want to manually install from an msi path, it seems weird that I'd need to provide both md5 and sha256 hashes.  It would be nice if I could do it with neither with just a warning.